src/handlers/issue-deduplication.ts

```diff
@@ -0,0 +1,138 @@
+import { IssueSimilaritySearchResult } from "../adapters/supabase/helpers/issues";
+import { Context } from "../types";
+const MATCH_THRESHOLD = 0.95;
+const WARNING_THRESHOLD = 0.5;
+
+export interface IssueGraphqlResponse {
+  node: {
+    title: string;
+    url: string;
+  };
+}
+
+/**
+ * Check if an issue is similar to any existing issues in the database
+ * @param context
+ * @returns true if the issue is similar to an existing issue, false otherwise
+ */
+export async function issueChecker(context: Context): Promise<boolean> {
+  const {
+    logger,
+    payload,
+    adapters: { supabase },
+  } = context;
+
+  const issue = payload.issue;
+
+  //First Check if an issue with more than MATCH_THRESHOLD similarity exists (Very Similar)
+  const similarIssue = await supabase.issue.findSimilarIssues(issue.body + issue.title, MATCH_THRESHOLD, issue.node_id);
+  if (similarIssue && similarIssue?.length > 0) {
+    logger.info(`Similar issue which matches more than ${MATCH_THRESHOLD} already exists`);
+    //Close the issue as "unplanned"
+    await context.octokit.issues.update({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      issue_number: issue.number,
+      state: "closed",
+      labels: ["unplanned"],
+    });
+    return true;
+  }
+
+  //Second Check if an issue with more than WARNING_THRESHOLD similarity exists (Warning)
+  const warningIssue = await supabase.issue.findSimilarIssues(issue.body + issue.title, WARNING_THRESHOLD, issue.node_id);
+  if (warningIssue && warningIssue?.length > 0) {
+    logger.info(`Similar issue which matches more than ${WARNING_THRESHOLD} already exists`);
+    //Add a comment immediately next to the issue
+    //Build a list of similar issues url
+    const issueList: IssueGraphqlResponse[] = await Promise.all(
+      warningIssue.map(async (issue: IssueSimilaritySearchResult) => {
+        //fetch the issue url and title using globalNodeId
+        const issueUrl: IssueGraphqlResponse = await context.octokit.graphql(
+          `query($issueNodeId: ID!) {
+                    node(id: $issueNodeId) {
+                        ... on Issue {
+                        title
+                        url
+                        }
+                    }
+                    }`,
+          {
+            issueNodeId: issue.issue_id,
+          }
+        );
+        return issueUrl;
+      })
+    );
+
+    // Reopen the issue
```

Comment:
What is going on here? Reopening issues is out of scope.

Reply:
Removed.